docs/api.md

```diff
@@ -118,8 +121,46 @@ type AddonsSpec struct {
 }
 
 // TODO maybe we have profiles for scaling behaviors
-type AutoScalerAddonSpec struct {
-    ResourceLimits LimitRange
+type ClusterAutoscaling struct {
+	// Maximum number of nodes in all node groups.
+	// Cluster autoscaler will not grow the cluster beyond this number.
+	// +kubebuilder:validation:Minimum=0
+	MaxNodesTotal int32 `json:"maxNodesTotal"`
+
+	// Should CA scale down the cluster
+    // default: true
+	ScaleDownEnabled bool `json:"enabled,omitempty"`
+
+	// Gives pods graceful termination time before scaling down
+    // default: 600 seconds
+	MaxPodGracePeriod *int32 `json:"maxPodGracePeriod,omitempty"`
+
+	// Maximum time CA waits for node to be provisioned
+    // default: 15 minutes
+	// +kubebuilder:validation:Pattern=^([0-9]+(\.[0-9]+)?(ns|us|µs|ms|s|m|h))+$
+	MaxNodeProvisionTime string `json:"maxNodeProvisionTime,omitempty"`
+
+	// To allow users to schedule "best-effort" pods, which shouldn't trigger
+	// Cluster Autoscaler actions, but only run when there are spare resources available,
+    // default: -10
+	// More info: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#how-does-cluster-autoscaler-work-with-pod-priority-and-preemption
+	PodPriorityThreshold *int32 `json:"podPriorityThreshold,omitempty"`
+
+	// BalanceSimilarNodeGroups enables/disables the
+	// `--balance-similar-node-groups` cluster-autocaler feature.
+	// This feature will automatically identify node groups with
+	// the same instance type and the same set of labels and try
+	// to keep the respective sizes of those node groups balanced.
+    // default: false
+	BalanceSimilarNodeGroups *bool `json:"balanceSimilarNodeGroups,omitempty"`
+
+	// Enables/Disables `--ignore-daemonsets-utilization` CA feature flag. Should CA ignore DaemonSet pods when calculating resource utilization for scaling down.
+    // default: false
+	IgnoreDaemonsetsUtilization *bool `json:"ignoreDaemonsetsUtilization,omitempty"`
```

Comment:
If these default to false, is there a need to have them as bool pointers? This is typically a hack to allow a nil entry to imply a true default

Reply:
i'm good with that

Reply:
> This is typically a hack to allow a nil entry to imply a true default

funnily enough, we use it in exactly the opposite manner in the CAO

Reply:
I propose removing this entirely and splitting out a separate discussion about a policy enum of some kind

Reply:
i chatted with Joel briefly about the pointer issue, and i'm going to leave these as pointers for the time being. the main reasoning here is that in the code we gate on a `nil` to decide if we should add the command line flag or not to the deployment spec for the autoscaler. we rely on the defaults that are part of the autoscaler binary instead of maintaining a set of defaults in the deployment code.

curious to hear if there are any other opinions about the pointers here, otherwise i will leave them as is.